runtime/src/main/java/org/corfudb/runtime/BootstrapUtil.java

```diff
@@ -66,12 +113,15 @@ public static void bootstrap(@Nonnull Layout layout,
                             .addClient(new ManagementHandler())
                             .addClient(new BaseHandler());
 
-                    new LayoutClient(router, layout.getEpoch())
-                            .bootstrapLayout(layout).get();
-                    new ManagementClient(router, layout.getEpoch())
-                            .bootstrapManagement(layout).get();
+                    bootstrapLayoutServer(router, layout);
+                    bootstrapManagementServer(router, layout);
+
                     router.stop();
                     break;
+                } catch (AlreadyBootstrappedException abe) {
+                    log.error("Bootstrapping node:{} failed with exception:", server, abe);
+                    log.error("Cannot retry since already bootstrapped.");
+                    throw new RuntimeException(abe);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)